File: Projekt/API/API/Models/Comment.cs
using System;

public class Comment
{
    public Comment(){}
    public Comment(CommentDto dto)
    {
        Id = dto.Id;
        Text = dto.Text;
        CreatedAt = DateTime.ParseExact(dto.CreatedAt, "yyyy-MM-dd HH:mm:ss", System.Globalization.CultureInfo.InvariantCulture);
    }
    public int Id { get; set; }
    public string Text { get; set; }
    public DateTime CreatedAt { get; set; }
    public UserProfile User { get; set; }
    public int PostId { get; set; }
}

File: Projekt/mobi_app/lib/models/Comment.dart
import 'package:json_annotation/json_annotation.dart';
import 'UserProfile.dart';

part 'Comment.g.dart';

@JsonSerializable()
class Comment {
  int id;
  String text;
  DateTime createdAt;
  UserProfile? user;
  int postId;

  Comment(this.id, this.text, this.createdAt, this.user, this.postId,);

  factory Comment.fromJson(Map<String, dynamic> json) =>
      _$CommentFromJson(json);

  Map<String, dynamic> toJson() => _$CommentToJson(this);
}